esmvalcore/_recipe.py

```diff
@@ -433,9 +433,25 @@ def _get_fx_files(variable, fx_info, config_user):
         fx_files = _get_input_files(fx_info, config_user)[0]
 
     # Flag a warning if no files are found
+    # CMIP6 special case:
+    # Before completely giving up, try looking for fx files in piControl:
+    # a lot of CMIP6 models have fx data only in piControl
     if not fx_files:
         logger.warning("Missing data for fx variable '%s'",
                        fx_info['short_name'])
+        if fx_info['project'] == 'CMIP6':
+            logger.debug(
+                "Missing data for fx variable %s "
+                "and experiment %s; since CMIP6, "
+                "will try experiment piControl, most fx "
+                "stuff is there.", fx_info['short_name'], fx_info['exp'])
+            fx_info['exp'] = 'piControl'
```

Comment:
```suggestion
            fx_info = dict(fx_info)
            fx_info['exp'] = 'piControl'
```

To avoid unwanted side effects.

Reply:
why do you want a copy of that, it'll get destroyed at return point anyway?

Reply:
ok am silly, that actually gets returned, good call!

Reply:
`dict` are mutable, so of you change that in the function it's also changed in the code the calls the function,